pdns/remote_logger.hh

```diff
@@ -113,17 +146,16 @@ private:
   {
     CircularWriteBuffer d_writer;
     std::unique_ptr<Socket> d_socket{nullptr};
+    RemoteLoggerInterface::Stats d_stats{};
   };
 
   ComboAddress d_remote;
-  std::atomic<uint64_t> d_drops{0};
-  std::atomic<uint64_t> d_processed{0};
   uint16_t d_timeout;
   uint8_t d_reconnectWaitTime;
   std::atomic<bool> d_exiting{false};
   bool d_asyncConnect{false};
 
-  LockGuarded<RuntimeData> d_runtime;
+  mutable LockGuarded<RuntimeData> d_runtime;
```

Comment:
I'm worried about making this mutable, this is fine now since we are only using that to acquire the lock to read metrics, but it means we could later alter something more critical in a `const` method and not notice. I wonder how painful it would be to make `getStats()` and `toString()` not `const` instead..

Reply:
Fixed as suggested.